### PR TITLE
IFTAQCAB-52 Verifying speciality delete feature

### PR DESCRIPTION
**Pull request by promahos**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 
 import static com.codeborne.selenide.Condition.appear;
 import static com.codeborne.selenide.Condition.disappear;
+import static com.codeborne.selenide.Condition.exactText;
+import static com.codeborne.selenide.Selenide.$$x;
 import static com.codeborne.selenide.Selenide.$x;
 
 public class SpecialitiesPage extends MainMenu {
 
+    private SelenideElement buttonNavigationLast = $x("//button[contains(@class,'paginator-navigation-last')]");
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:15]: create NAVIGATION_BUTTON_PATH_TEMPLATE: "//button[contains(@class,'paginator-navigation-%s')]"

Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:15]: and use format(NAVIGATION_BUTTON_PATH_TEMPLATE, "last")
format(NAVIGATION_BUTTON_PATH_TEMPLATE, "first")
format(NAVIGATION_BUTTON_PATH_TEMPLATE, "next")

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 2]
```diff
@@ -46,4 +61,48 @@ public String getMessageText() {
                 .should(appear)
                 .getText();
     }
+
+    @Step("Speciality page: Deleted speciality by code")
+    public SpecialitiesPage deleteSpeciality(SpecialityEntity speciality) {
+        SelenideElement currentRow = findRowByColumnValue(speciality.getCode()); //consider code as unique field
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:67]: var

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 3]
```diff
@@ -46,4 +61,48 @@ public String getMessageText() {
...
+            currentRow = findRowOnCurrentPageByColumnValue(searchValue);
+        }
+
+        return currentRow;
+    }
+
+    private SelenideElement findRowOnCurrentPageByColumnValue(String searchValue) {
+        SelenideElement searchRow = null;
+        var tableRows = $$x("//table//tr//td");
+        if (tableRows
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:98]: add new line before 'if'

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 4]
```diff
@@ -46,4 +61,48 @@ public String getMessageText() {
...
+    }
+
+    private SelenideElement findRowByColumnValue(String searchValue) {
+        SelenideElement currentRow;
+
+        if (buttonNavigationFirst.isEnabled()) {
+            buttonNavigationFirst.click();
+        }
+
+        currentRow = findRowOnCurrentPageByColumnValue(searchValue);
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:81]: var currentRow = findRowOnCurrentPageByColumnValue(searchValue);

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 5]
```diff
@@ -46,4 +61,48 @@ public String getMessageText() {
...
+    public SpecialitiesPage deleteSpeciality(SpecialityEntity speciality) {
+        SelenideElement currentRow = findRowByColumnValue(speciality.getCode()); //consider code as unique field
+        currentRow.$x(".//i[contains(@class,'delete')]").click();
+        $x("//button/span[contains(text(),'Підтвердити')]").click();
+
+        return this;
+    }
+
+    private SelenideElement findRowByColumnValue(String searchValue) {
+        SelenideElement currentRow;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:75]: delete

File: src/main/java/com/softserveinc/ita/pageobjects/models/SpecialityEntity.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import static com.softserveinc.ita.pageobjects.util.RandomUtil.getRandomNumber;
+
+@Data
+public class SpecialityEntity {
+    private String code;
+    private String name;
+
+    public SpecialityEntity() {
+        var randCode = getRandomNumber(5);
+        this.code = Integer.toString(randCode); // only numbers, no more than 5 symbols;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/models/SpecialityEntity.java:16]: use valueOf() from String class instead of toString

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 7]
```diff
@@ -50,7 +64,7 @@ public SpecialitiesPage waitForProgressBarToDisappear() {
 
     @Step("Speciality page: Waited for progress bar to appear")
     public SpecialitiesPage waitForProgressBarToAppear() {
-        progressBar.should(appear);
+        progressBar.should(appear, Duration.ofSeconds(3));
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:67]: static import of 'ofSeconds'

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 8]
```diff
@@ -62,47 +76,50 @@ public String getMessageText() {
...
-            currentRow = findRowOnCurrentPageByColumnValue(searchValue);
+            isSearchValueOnCurrentPage = isSearchValueInTableTexts(searchValue);
         }
 
-        return currentRow;
+        return this;
+    }
+
+    @Step("Speciality page: Changed table page")
+    private void goToTablePage(String direction) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:99]: private methods should be placed after public ones

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java [hunk 9]
```diff
@@ -62,47 +76,50 @@ public String getMessageText() {
...
-    private SelenideElement findRowByColumnValue(String searchValue) {
-        SelenideElement currentRow;
-
-        if (buttonNavigationFirst.isEnabled()) {
-            buttonNavigationFirst.click();
-        }
-
-        currentRow = findRowOnCurrentPageByColumnValue(searchValue);
+        var buttonNavigationNext = $x(format(NAVIGATION_BUTTON_PATH_TEMPLATE, "next"));
+        boolean isSearchValueOnCurrentPage = false;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:84]: var

File: src/main/java/com/softserveinc/ita/steps/SpecialitiesSteps.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+public class SpecialitiesSteps {
+
+    private SpecialitiesPage page;
+
+    public void openSpecialityPage() {
+        page = new LoginPage()
+                .login(ADMIN_LOGIN, ADMIN_PASSWORD)
+                .openSpecialitiesPage();
+    }
+    public void addNewSpeciality(SpecialityEntity speciality) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/steps/SpecialitiesSteps.java:21]: add a new line between methods

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java (hunk 8 above)
Comment (promahos) [reply src/main/java/com/softserveinc/ita/pageobjects/admin/SpecialitiesPage.java:99]: It was supposed to be public( I'll fix

**Pull request merged**